bodo/tests/test_df_lib/test_end_to_end.py

```diff
@@ -2392,3 +2392,195 @@ def test_fallback_wrapper_deep_fallback():
     assert "TypeError triggering deeper fallback" in warning_msg, (
         f"Unexpected warning message: {warning_msg}"
     )
+
+
+import benchmarks.tpch.dataframe_lib as tpch
+from benchmarks.tpch.dataframe_lib import _query_to_datasets, collect_datasets
+
+datapath = "bodo/tests/data/tpch-test_data/parquet"
+
+
+def load_tpch_data(name, datapath):
+    if name == "lineitem":
+        return tpch.load_lineitem(datapath, pd=pd)
+    if name == "part":
+        return tpch.load_part(datapath, pd=pd)
+    if name == "orders":
+        return tpch.load_orders(datapath, pd=pd)
+    if name == "customer":
+        return tpch.load_customer(datapath, pd=pd)
+    if name == "supplier":
+        return tpch.load_supplier(datapath, pd=pd)
+    if name == "nation":
+        return tpch.load_nation(datapath, pd=pd)
+    if name == "region":
+        return tpch.load_region(datapath, pd=pd)
+    if name == "partsupp":
+        return tpch.load_partsupp(datapath, pd=pd)
+    raise KeyError("Invalid tpch data name {name}.")
+
+
+def run_tpch_query_test(query_func, num, is_dataframe):
+    pd_args = [load_tpch_data(key, datapath) for key in _query_to_datasets[num]]
+    bd_args = [bd.from_pandas(df) for df in pd_args]
+
+    pd_kwargs = {"pd": pd}
+    pd_result = query_func(*pd_args, **pd_kwargs)
+    bd_result = query_func(*bd_args)
+
+    if is_dataframe:
+        bd_result.is_lazy_plan()
+        _test_equal(bd_result, pd_result, check_pandas_types=False, reset_index=True)
+    else:
+        # For scalar or numeric results
+        assert isinstance(pd_result, (int, float)) and isinstance(
+            bd_result, (int, float)
+        )
+        assert np.isclose(pd_result, bd_result)
+
+
+def test_tpch_q01():
+    collect_datasets(tpch.q01)
```

Comment:
I think this line isn't needed.  It is just doing the query_to_dataset that your run_tpch_query_test is already doing.  Repeat for all tests below.